### ISSUE TITLE
Using `Class.constructor.name` as identifier to register and refer to components is leading to issues with bundlers and code minification

### ISSUE THREAD

**Issue by martinemmert** [fritzy/ape-ecs#49]
Scenario:
- using Webpack 5
- using PIXI.js as Dependency

Problem Description:

When Webpack bundles all packages together conflicting declaration names of functions/classes are renamed to solve this issue without requiring the developer to act. The applied resolution is: one of the declaration names remains as it is and the other ones will be renamed to something like `${declaration name}__${declaration name}`.

Since Pixi is used as a dependency, it is inserted before my code in the final bundle. Therefore my `DisplayObject` component class is renamed to `DisplayObject_DisplayObject` which is then of course registered as `DisplyObject_DisplayObject` inside the worlds component registry.

This leads to the error `... DisplayObject is not registered ...` while creating entities.

**Note:** 
This error first came up while running the project in production mode without modification.
Therefore I suggest the tests should include testing of minified and bundled code.

---

I am currently working on a solution, but I'd like to test a bit further.
[Branch](https://github.com/martinemmert/ape-ecs/tree/feat/component/add-static-display-name-prop) | [commit of working solution](https://github.com/martinemmert/ape-ecs/commit/698f7907216c553974d38fc46136222b08743f16)

**Comment by fritzy** [comment]
Ah, that is problematic. Okay, I'll add bundler tests, and I'll look to fix this specific issue.

**Comment by martinemmert** [comment]
I already came up with a solution and used it without any trouble so far. But, I did not write any new tests for that — That's why I did not come up with a pull request until yet.

https://github.com/martinemmert/ape-ecs/commits/feat/component/add-static-display-name-prop

This is the current branch I am using. I am focused more on building the engine for my game currently but will come back to this soon.
